src/smt/sygus_solver.cpp

```diff
@@ -431,9 +431,13 @@ void SygusSolver::initializeSygusSubsolver(std::unique_ptr<SolverEngine>& se,
                                            Assertions& as)
 {
   initializeSubsolver(se, d_env);
+  std::unordered_set<Node> processed;
+  // if we did not spawn a subsolver for the main check, the overall SyGuS
+  // conjecture has been added as an assertion. Do not add it here, which
+  // is important for check-synth-sol.
```

Comment:
clarify that it's not an issue to add it to `processed` also when we are not spawning subsolvers for the main check.